File: app/src/main/java/com/sdex/activityrunner/intent/dialog/ValueInputDialog.java
package com.sdex.activityrunner.intent.dialog;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.EditText;
import com.sdex.activityrunner.R;
import com.sdex.activityrunner.util.ObjectsCompat;

public class ValueInputDialog extends DialogFragment {

  public static final String TAG = "ValueInputDialog";

  private static final String ARG_TYPE = "arg_type";
  private static final String ARG_INITIAL_VALUE = "arg_initial_value";

  private OnValueInputDialogCallback callback;

  public static ValueInputDialog newInstance(int type, String initialValue) {
    Bundle args = new Bundle(2);
    args.putInt(ARG_TYPE, type);
    args.putString(ARG_INITIAL_VALUE, initialValue);
    ValueInputDialog fragment = new ValueInputDialog();
    fragment.setArguments(args);
    return fragment;
  }

  @NonNull
  @Override
  public Dialog onCreateDialog(Bundle savedInstanceState) {
    final int type;
    final String initialValue;
    if (getArguments() != null) {
      type = getArguments().getInt(ARG_TYPE);
      initialValue = getArguments().getString(ARG_INITIAL_VALUE, "");
    } else {
      throw new NullPointerException();
    }
    final AlertDialog.Builder builder =
      new AlertDialog.Builder(ObjectsCompat.requireNonNull(getActivity()));
    View view = View.inflate(getActivity(), R.layout.dialog_input_value, null);
    final EditText valueView = view.findViewById(R.id.value);
    valueView.setText(initialValue);
    valueView.setSelection(initialValue.length());
    builder.setTitle(type)
      .setView(view)
      .setPositiveButton(android.R.string.ok, (dialog, which) -> {
        final String newValue = valueView.getText().toString();
        callback.onValueSet(type, newValue);
      })
      .setNegativeButton(android.R.string.cancel, null);
    return builder.create();
  }

  @Override
  public void onAttach(Context context) {
    super.onAttach(context);
    try {
      callback = (OnValueInputDialogCallback) context;
    } catch (ClassCastException e) {
      throw new ClassCastException(context.toString()
        + " must implement OnValueInputDialogCallback");
    }
  }

  public interface OnValueInputDialogCallback {

    void onValueSet(int type, String value);
  }
}

File: app/src/main/java/com/sdex/activityrunner/intent/dialog/ValueInputDialog.kt
package com.sdex.activityrunner.intent.dialog

import android.app.Dialog
import android.os.Bundle
import android.view.inputmethod.EditorInfo
import androidx.core.os.bundleOf
import androidx.fragment.app.activityViewModels
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.sdex.activityrunner.commons.BaseDialogFragment
import com.sdex.activityrunner.databinding.DialogInputValueBinding
import com.sdex.activityrunner.intent.LaunchParamsViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class ValueInputDialog : BaseDialogFragment() {

    private val viewModel: LaunchParamsViewModel by activityViewModels()

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val type = requireArguments().getInt(ARG_TYPE)
        val initialValue = requireArguments().getString(ARG_INITIAL_VALUE, "")

        val binding = DialogInputValueBinding.inflate(requireActivity().layoutInflater)

        binding.valueView.setText(initialValue)
        binding.valueView.setSelection(initialValue.length)
        binding.valueView.setOnEditorActionListener { _, actionId, _ ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                val newValue = binding.valueView.text.toString()
                viewModel.setValue(type, newValue)
                dismiss()
                return@setOnEditorActionListener true
            }
            false
        }
        return MaterialAlertDialogBuilder(requireActivity())
            .setTitle(type)
            .setView(binding.root)
            .setPositiveButton(android.R.string.ok) { _, _ ->
                val newValue = binding.valueView.text.toString()
                viewModel.setValue(type, newValue)
            }
            .setNegativeButton(android.R.string.cancel, null)
            .create()
    }

    companion object {

        const val TAG = "ValueInputDialog"

        private const val ARG_TYPE = "arg_type"
        private const val ARG_INITIAL_VALUE = "arg_initial_value"

        fun newInstance(type: Int, initialValue: String?): ValueInputDialog {
            return ValueInputDialog().apply {
                arguments = bundleOf(
                    ARG_TYPE to type,
                    ARG_INITIAL_VALUE to initialValue
                )
            }
        }
    }
}
